internal/app/client/client.go

```diff
@@ -119,26 +140,25 @@ func invoke(ctx context.Context, req *v1alpha1.ClientCompatRequest) (*v1alpha1.C
 		clientOptions = append(clientOptions, connect.WithProtoJSON())
 	}
 
-	// TODO - How do we configure each compression algo? i.e.
-	// how do we know the string to use and the func for WithCompression?
+	// TODO - Add support for other compression algos
 	switch req.Compression {
 	case v1alpha1.Compression_COMPRESSION_GZIP:
 		clientOptions = append(clientOptions, connect.WithSendGzip())
-	case v1alpha1.Compression_COMPRESSION_IDENTITY:
 	case v1alpha1.Compression_COMPRESSION_BR:
 	case v1alpha1.Compression_COMPRESSION_ZSTD:
 	case v1alpha1.Compression_COMPRESSION_DEFLATE:
 	case v1alpha1.Compression_COMPRESSION_SNAPPY:
+	case v1alpha1.Compression_COMPRESSION_IDENTITY:
```

Comment:
We should be returning an error for the unsupported compression algorithms instead of doing nothing.

Also, these all happen to do nothing, but not because they fall-through. There is no fall-through in Go switch statements unless you do so explicitly with the `fallthrough` keyword.

The preferred way to instead have multiple values map to the same case is to use comma-separate list of case values:
```suggestion
	case v1alpha1.Compression_COMPRESSION_BR, v1alpha1.Compression_COMPRESSION_ZSTD.
		v1alpha1.Compression_COMPRESSION_DEFLATE, v1alpha1.Compression_COMPRESSION_SNAPPY:
		// return an error since we don't actually support any of these yet
	case v1alpha1.Compression_COMPRESSION_IDENTITY, v1alpha1.Compression_COMPRESSION_UNSPECIFIED:
		// do nothing
```